File: Lib/Telerik/Examples/QuickStart/GridView/Rows/RowsDragDrop/CustomGridBehavior.vb
Imports Telerik.WinControls.UI

Namespace Telerik.Examples.WinControls.GridView.Rows.RowsDragDrop
	Public Class CustomGridBehavior
		Inherits BaseGridBehavior
'INSTANT VB NOTE: The variable selectedRows was renamed since Visual Basic does not allow class members with the same name:
		Private selectedRows_Renamed As New List(Of GridViewRowInfo)()

		Public ReadOnly Property SelectedRows() As List(Of GridViewRowInfo)
			Get
				Return Me.selectedRows_Renamed
			End Get
		End Property

		Public Overrides Function OnMouseDown(ByVal e As MouseEventArgs) As Boolean
			selectedRows_Renamed.Clear()
			Dim result As Boolean = MyBase.OnMouseDown(e)

			If (Control.ModifierKeys And Keys.Control) = Keys.Control OrElse (Control.ModifierKeys And Keys.Shift) = Keys.Shift Then
				selectedRows_Renamed.AddRange(Me.GridControl.SelectedRows)
			Else
				selectedRows_Renamed.Add(Me.GridControl.CurrentRow)
			End If

			Return result
		End Function
	End Class
End Namespace

File: Lib/Telerik/Examples/QuickStart/GridView/Rows/RowsDragDrop/CustomGridBehavior.cs
using System.Collections.Generic;
using System.Windows.Forms;
using Telerik.WinControls.UI;

namespace Telerik.Examples.WinControls.GridView.Rows.RowsDragDrop
{
    public class CustomGridBehavior : BaseGridBehavior
    {
        List<GridViewRowInfo> selectedRows = new List<GridViewRowInfo>();

        public List<GridViewRowInfo> SelectedRows
        {
            get { return this.selectedRows; }
        }

        public override bool OnMouseDown(MouseEventArgs e)
        {
            selectedRows.Clear();
            bool result = base.OnMouseDown(e);

            if ((Control.ModifierKeys & Keys.Control) == Keys.Control ||
                (Control.ModifierKeys & Keys.Shift) == Keys.Shift)
            {
                selectedRows.AddRange(this.GridControl.SelectedRows);
            }
            else
            {
                selectedRows.Add(this.GridControl.CurrentRow);
            }

            return result;
        }
    }
}
